scripts/ci/create-docker-image.sh

```diff
@@ -6,77 +6,78 @@ if [ -z "$REF" ]; then
 fi
 
 set -v
+set -x
 set -e
 
-ARCHES="amd64 ppc64le"
-TAG=`echo $REF | awk -F '/' '{print $NF}'`
-VERSION=`echo $TAG | tr -d [a-z]`
-DOCKER_IMAGE=skydive/skydive
-DOCKER_EMAIL=skydivesoftware@gmail.com
-DOCKER_USERNAME=skydiveproject
-[ -n "$VERSION" ] && DOCKER_TAG=$VERSION || DOCKER_TAG=latest
+# Arches are the docker arch names
+: ${ARCHES:=amd64 ppc64le s390x}
+# arm64 waiting on  golang 1.11 (https://github.com/skydive-project/skydive/pull/1188#discussion_r204336060)
+: ${DOCKER_IMAGE:=skydive/skydive}
+: ${DOCKER_USERNAME:=skydiveproject}
 
-function cleanup {
-    docker rm -f skydive-crosscompile
-}
+TAG=${REF##*/}
+VERSION=${TAG//[a-z]/}
 
-function cross_compile() {
-    docker pull ubuntu:18.04
-    docker run -tid --name skydive-crosscompile -v `pwd`:/root/go/src/github.com/skydive-project/skydive -v $GOPATH/.cache/govendor:/root/go/.cache/govendor ubuntu:18.04 /bin/bash
-    trap cleanup ERR
-    docker exec ${http_proxy:+--env http_proxy=$http_proxy} skydive-crosscompile /root/go/src/github.com/skydive-project/skydive/scripts/ci/create-docker-multiarch-image.sh $1 $2 $3
-    docker build -t ${DOCKER_IMAGE}:${arch}-${DOCKER_TAG} -f contrib/docker/Dockerfile.${arch} contrib/docker/
-    docker rm -f skydive-crosscompile
-}
+[ -n "$VERSION" ] && DOCKER_TAG=$VERSION || DOCKER_TAG=latest
 
-for arch in $ARCHES
-do
-  case $arch in
-    amd64)
-      # x86_64 image
-      make docker-image WITH_EBPF=true DOCKER_IMAGE=${DOCKER_IMAGE} DOCKER_TAG=amd64-${DOCKER_TAG}
-      ;;
-    ppc64le)
-      cross_compile powerpc64le ppc64le ppc64el
-      ;;
-    \?)
-      echo "Unsupported architecture $arch" >&2
-      exit 1
-      ;;
-  esac
-done
+# See if a server forms part of DOCKER_IMAGE, e.g. DOCKER_IMAGE=registry.ng.bluemix.net:8080/skydive/skydive
+if [[ "$DOCKER_IMAGE" =~ /[^/]*/[^/]* ]]; then
+    DOCKER_SERVER=${DOCKER_IMAGE%/[^/]*/[^/]*}
+fi
+
+if [ -n "$PUSH_RUN" ]; then
+    echo "Running in push run mode. Skipping build."
+else
+    for arch in $ARCHES
+    do
+        case $arch in
+          amd64)
+            # x86_64 image
+            make docker-build DOCKER_IMAGE=${DOCKER_IMAGE} DOCKER_TAG=amd64-${DOCKER_TAG}
```

Comment:
I would prefer to keep using make docker-image as it is way faster (no need to start the docker container and install all the dependencies)

Reply:
Otherwise, it's really nice :-)

Do you think we can close this one : https://github.com/skydive-project/skydive/pull/1188 ?

Reply:
> I would prefer to keep using make docker-image as it is way faster...

I noticed the jenkins doesn't seem to preserve any of the docker base images, or build stages, skydive-{cross}compile images, or govendor-cache volumes all of which which would speed it up.

Sure I can drop this back to a `docker-image` target. I was planning to use the same script within IBM but with `ARCHES` / `DRY_RUN` / `PUSH_RUN` I can work around with it.

Last thought - should the `skydive.$ARCH` executable be stripped before going in the docker image? 

Thanks for tolerating my updates on this in the last few days and thanks for the positive review.

Reply:
One problem with Ubuntu is that we can't statically build Skydive with the OpenContrail probe.
- If you're ok, I suggest we don't put a static binary into the image (we can't them replace the libpcap installation with an `apt install libpcap` and build with WITH_OPENCONTRAIL=true WITH_EBPF=true
- use ubuntu as base image for all our containers

Reply:
added commit to revert to `docker-image` for amd64

removed pcap install and left to libpcap packages. These are staticly linked so shouldn't need a package installed on final image.

base non-x86_64 images changed to ubuntu default (overridable with `$BASE`). Ubuntu needed to be quantified with $arch as its a cross-build base image.

I don't follow the static requirement being incompatible with OpenContrail. The OpenContrail seems to be http webservice or a file lookin at https://github.com/nlewo/contrail-introspect-cli). I'm sure I'm probably missing something.